package.json

```diff
@@ -27,18 +27,21 @@
     "lint-staged": "^9.5.0",
     "lodash": "^4.17.15",
     "prettier": "^1.19.1",
+    "snyk": "^1.269.0",
     "typescript": "^3.7.4"
   },
   "scripts": {
     "build": "lerna run build",
     "coverage:publish": "codecov && lerna run coverage:publish -- -f coverage/lcov.info",
     "lint": "lerna run lint",
-    "release": "lerna version --conventional-commits",
+    "preversion": "snyk protect",
```

Comment:
So it's now `preversion`instead of `prepublish`?

Reply:
🤔 I'm not familiar with Lerna life cycle ... let's wait @sergiohgz

Reply:
Because we manage automatic release from CI, but versioning is made locally, so you cannot update version unless Snyk pass

Reply:
But I guess this changes the workflow.

Reply:
@juanpicado that's why I have mention @lirantal, because I don't know if we need to run command for every package independently or we can run for all in one. 
If we need to run independently, we will change the command invoked in preversion script with `lerna exec snyk protect` (as I explained in [this article](https://verdaccio.org/blog/2019/09/07/managing-multiples-projects-with-lerna-and-yarn-workspaces#running-scripts) :stuck_out_tongue_winking_eye:)

Reply:
No @DanielRuf, this change should not change the workflow. Opposite, if we extract google-cloud package from monorepo, we will not have snyk installed here, so it's better to have it in the root.

Reply:
Well, `preversion` and `prepublish` are still two different workflow steps for me =)

Reply:
Aaah, I understand your point of view. Of course, both are different scripts and they are performed in different situations.
Before this PR, if you update versions (using the scripts defined in root package.json) and a google-cloud dependency has known security issues, CI will publish some packages before fail with google-cloud due to snyk command (due to topological order, some examples are eslint-config or babel-preset).
If we change with preversion, this problem will happen in our local machines, before publish any package in CI, so we can perform a fix before making invalid releases.

Reply:
Ok, thanks for clarifying this. Would be good to add this to the PR description / commit message too. Just to know later why it was changed to this (the reasoning behind it).

Reply:
Good idea, let me some time to organize better and I will improve commit message :wink:

Reply:
@DanielRuf I have improved commit message and add the same in PR description.

Reply:
Also, here is the trace of performing `yarn release` to generate a new version (with `--no-push --yes` options to not make a dry run), where we can see that it works like expected:
```
[sergio@f31-sergiohgz monorepo]$ yarn release --no-push --yes
yarn run v1.21.1
$ lerna version --conventional-commits --no-push --yes
lerna notice cli v3.20.1
lerna info current version 8.5.3
lerna info Looking for changed packages since v8.5.3
lerna info getChangelogConfig Successfully resolved preset "conventional-changelog-angular"

Changes:
 - @verdaccio/commons-api: 8.5.3 => 9.0.0
 - @verdaccio/file-locking: 8.5.3 => 9.0.0
 - @verdaccio/readme: 8.5.3 => 9.0.0
 - @verdaccio/streams: 8.5.3 => 9.0.0
 - @verdaccio/types: 8.5.2 => 9.0.0
 - @verdaccio/active-directory: 8.5.3 => 9.0.0
 - verdaccio-audit: 8.5.3 => 9.0.0
 - verdaccio-auth-memory: 8.5.3 => 9.0.0
 - verdaccio-aws-s3-storage: 8.5.3 => 9.0.0
 - verdaccio-google-cloud: 8.5.3 => 9.0.0
 - verdaccio-htpasswd: 8.5.3 => 9.0.0
 - @verdaccio/local-storage: 8.5.3 => 9.0.0
 - verdaccio-memory: 8.5.3 => 9.0.0
 - @verdaccio/babel-preset: 8.5.3 => 9.0.0
 - @verdaccio/eslint-config: 8.5.3 => 9.0.0
 - eslint-plugin-verdaccio: 8.5.3 => 9.0.0
 - @verdaccio/testing-utilities: 8.5.3 => 9.0.0

lerna info auto-confirmed
lerna info execute Skipping git push
lerna info execute Skipping releases
lerna info lifecycle @verdaccio/monorepo@undefined~preversion: @verdaccio/monorepo@undefined
lerna WARN lifecycle The node binary used for scripts is /tmp/yarn--1577803403219-0.5603776510190481/node but npm is using /usr/local/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> @verdaccio/monorepo@undefined preversion /home/sergio/Documentos/github/verdaccio/monorepo
> snyk protect

Successfully applied Snyk patches

lerna success version finished
Done in 88.06s.
```

Reply:
Sorry for jumping late on this. I'm still catching up on the thread but want to point out some things:
- We have improved support for monorepos, especially with the latest release of the CLI version 1.266.0 so things might behave better now
- generally speaking, applying patches with snyk protect makes sense for projects which actually ship out the vulnerable JS, so something like a frontend project that bundles libraries that makes sense, or a Node.js microservice - makes sense. Things like a generic library it doesn't make sense because patching the lib locally doesn't ensure the consumer won't be vulnerable for it when they install your library (due to no lockfile on the end user install).

If this repo is then occupying libs then snyk protect here doesn't make a whole lot of sense.